cmd/snap/cmd_advise.go

```diff
@@ -205,11 +210,50 @@ func adviseViaAptHook() error {
 	return nil
 }
 
+type Snap struct {
+	Snap string
+	Version string
+	Command string
+}
+
+func dumpDbHook() error {
+	commands, err := advisor.DumpCommands()
+	if err != nil {
+		return err
+	}
+
+	commands_processed := make([]string, 0, len(commands))
```

Comment:
you're sizing it _just_ too small to avoid growing it, now, which is the worst case. I'd suggest either leaving it alone and letting go do all the sizing (this is not a long-lived process, after all), or making it `len(commands)*11/10`